opendatahub-io/odh-dashboard
# feat: Modify NIM enablement process

https://issues.redhat.com/browse/NVPE-22
## Description


Change the enablement process for NIM on the dashboard.

## How Has This Been Tested?



Tested locally.

## Test Impact



- Go to the Applications -> Explore, and click the tile for NIM.
- On the right side panel, there should be an enable button.
- Click the enable button,  a modal should be shown. 
- Enter the API key provided by NVIDIA, and click the submit button. 
- A processor should be shown to indicate the validation process.
- If validation is successful, the modal should be closed, and a popup showing the success message should be displayed on the right-top corner. The odh-nim-account CR should be generated, and related resources should be generated as well.
- Go to Application -> Enabled, the NIM tile should be there. 
- If validation fails, error message should be displayed in the modal and in the popup at the right-top corner.

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 

<img width="1725" alt="image" src="https://github.com/user-attachments/assets/5fede496-e6d3-4bac-a659-ce0491dda584">
<img width="1712" alt="image" src="https://github.com/user-attachments/assets/d555de3c-5164-485d-954a-0b39f51e8198">
<img width="1697" alt="image" src="https://github.com/user-attachments/assets/d7079a05-d562-4ca9-b718-1e80e876a936">
<img width="1716" alt="image" src="https://github.com/user-attachments/assets/bc63d3ac-31ba-4272-8fb1-b665867e1764">
<img width="1692" alt="image" src="https://github.com/user-attachments/assets/2a63948a-7968-424a-b7e6-d775049190ad">


After the PR is posted & before it merges:
